File: server/db/store/library.go
package store

import (
	"github.com/dxtym/yomu/server/db/models"
)

func (s *Store) AddLibrary(record *models.Library) error {
	if err := s.db.Create(record).Error; err != nil {
		return err
	}

	return nil
}

func (s *Store) GetLibrary(userId int64) ([]models.Library, error) {
	var library []models.Library
	if err := s.db.Where("user_id = ?", userId).Find(&library).Error; err != nil {
		return nil, err
	}

	return library, nil
}

func (s *Store) RemoveLibrary(userId int64, manga string) error {
	return s.db.Where("user_id = ? AND manga = ?", userId, manga).Delete(&models.Library{}).Error
}

File: client/src/app/api/library.ts
import ApiClient from "./client";
import { IManga } from "@/types/manga";

export async function getLibrary(this: ApiClient): Promise<IManga[]> {
  const res = await this.client.get<IManga[]>("/library");
  return res.data;
}

export async function addLibrary(
  this: ApiClient, 
  manga: string, 
  coverImage: string
): Promise<void> {
  await this.client.post("/library", {
    manga: manga,
    cover_image: coverImage,
  });
}

export async function removeLibrary(this: ApiClient, manga: string): Promise<void> {
  await this.client.delete("/library", {
    data: { manga: manga },
  });
}
